### PR TITLE
Enable a few tests on netcore by removing PrivateObject

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenByDeviceCodeHandler.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public AcquireTokenByDeviceCodeHandler(RequestData requestData, DeviceCodeResult
             this.deviceCodeResult = deviceCodeResult;
         }
 
-        protected override async Task<AdalResultWrapper> SendTokenRequestAsync()
+        protected internal /* internal for test only */ override async Task<AdalResultWrapper> SendTokenRequestAsync()
```
Comment (bgavrilMS) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenByDeviceCodeHandler.cs:48]: I had to expose a few methods in order to move away from PrivateObject.

File: adal/tests/Test.ADAL.NET.Unit.net45/Helpers/PlatformParametersFactory.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+namespace Test.ADAL.NET.Unit
+{
+    public class PlatformParametersFactory
+    {
+        public static IPlatformParameters CreateDefault()
+        {
+#if DESKTOP
+            return new PlatformParameters(PromptBehavior.Auto);
+#elif NET_CORE
+            return new PlatformParameters();
```
Comment (bgavrilMS) [inline adal/tests/Test.ADAL.NET.Unit.net45/Helpers/PlatformParametersFactory.cs:13]: In the end this wasn't needed, as all tests that create PP invoke the interactive flow which is not available on netcore. But still, it's a good basis for tests in other platforms.

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenHandlerBase.cs [hunk 3]
```diff
@@ -307,7 +307,7 @@ protected virtual async Task PostTokenRequestAsync(AdalResultWrapper resultEx)
 
         protected abstract void AddAditionalRequestParameters(DictionaryRequestParameters requestParameters);
 
-        protected virtual async Task<AdalResultWrapper> SendTokenRequestAsync()
+        protected internal /* interal for test only */ virtual async Task<AdalResultWrapper> SendTokenRequestAsync()
```
Comment (jennyf19) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenHandlerBase.cs:310]: sp: internal

File: adal/tests/Test.ADAL.NET.Unit.net45/InstanceDiscoveryTests.cs [hunk 4]
```diff
@@ -277,17 +276,16 @@ public async Task TestInstanceDiscovery_WhenMetadataIsReturned_ShouldUsePreferre
                 }
             });
 
-            var privateObject = new PrivateObject(new AcquireTokenForClientHandler(new RequestData
```
Comment (jennyf19) [inline adal/tests/Test.ADAL.NET.Unit.net45/InstanceDiscoveryTests.cs:280]: >PrivateObject [](start = 36, length = 13)

what was PrivateObject for?

**Comment by jennyf19** [comment]
> using Microsoft.IdentityModel.Clients.ActiveDirectory;

Add header info

---
Refers to: adal/tests/Test.ADAL.NET.Unit.net45/Helpers/PlatformParametersFactory.cs:1 in a53c5e0. [](commit_id = a53c5e0656294680b71e18b6825f58bbac002f1e, deletion_comment = False)

File: msal/tests/Test.MSAL.NET.Unit.net45/JsonWebTokenTests.cs [hunk 5]
```diff
@@ -25,7 +25,9 @@
 //
 //------------------------------------------------------------------------------
 
-#if !NET_CORE
+//TODO: bogavril - raise a bug as these fail on netcore
```
Comment (jennyf19) [inline msal/tests/Test.MSAL.NET.Unit.net45/JsonWebTokenTests.cs]: >//TODO: [](start = 0, length = 7)

Is this still a TODO ? :)

**Review by jennyf19 — APPROVED** [review]
just a few questions/comments.

File: msal/tests/Test.MSAL.NET.Unit.net45/JsonWebTokenTests.cs (hunk 5 above)
Comment (bgavrilMS) [reply msal/tests/Test.MSAL.NET.Unit.net45/JsonWebTokenTests.cs]: Bug logged and comment changed.

**Pull request merged**